src/plugincollection.js

```diff
@@ -58,48 +70,89 @@ export default class PluginCollection {
 	/**
 	 * Loads a set of plugins and add them to the collection.
 	 *
-	 * @param {Function[]} plugins An array of {@link module:core/plugin~Plugin plugin constructors}.
+	 * @param {Array.<module:core/plugin~Plugin|String>} plugins An array of {@link module:core/plugin~Plugin plugin constructors}
+	 * or plugin names.
+	 * @param {Array.<String>} removePlugins
 	 * @returns {Promise} A promise which gets resolved once all plugins are loaded and available into the
 	 * collection.
-	 * @param {Array.<module:core/plugin~Plugin>} returns.loadedPlugins The array of loaded plugins.
+	 * @returns {Array.<module:core/plugin~Plugin>} returns.loadedPlugins The array of loaded plugins.
 	 */
-	load( plugins ) {
+	load( plugins, removePlugins = [] ) {
 		const that = this;
 		const editor = this._editor;
 		const loading = new Set();
 		const loaded = [];
 
+		// Plugins which should be remove can be the constructors or plugin names.
+		// We need to unify this because we are supporting loading plugins by names or plugin constructors.
+		removePlugins = removePlugins.reduce( ( arr, PluginConstructorOrName ) => {
+			arr.push( PluginConstructorOrName );
+
+			if ( typeof PluginConstructorOrName === 'string' ) {
+				arr.push( getPluginConstructor( PluginConstructorOrName ) );
+			} else {
+				arr.push( PluginConstructorOrName.pluginName );
+			}
+
+			return arr;
+		}, [] );
+
 		return Promise.all( plugins.map( loadPlugin ) )
 			.then( () => loaded );
 
-		function loadPlugin( PluginConstructor ) {
+		function loadPlugin( PluginConstructorOrName ) {
+			// Don't load the plugin if it cannot be loaded.
+			if ( removePlugins.includes( PluginConstructorOrName ) ) {
+				return;
+			}
+
 			// The plugin is already loaded or being loaded - do nothing.
-			if ( that.get( PluginConstructor ) || loading.has( PluginConstructor ) ) {
+			if ( that.get( PluginConstructorOrName ) || loading.has( PluginConstructorOrName ) ) {
 				return;
 			}
 
-			return instantiatePlugin( PluginConstructor )
+			return instantiatePlugin( PluginConstructorOrName )
 				.catch( ( err ) => {
 					/**
 					 * It was not possible to load the plugin.
 					 *
 					 * @error plugincollection-load
 					 * @param {String} plugin The name of the plugin that could not be loaded.
 					 */
-					log.error( 'plugincollection-load: It was not possible to load the plugin.', { plugin: PluginConstructor } );
+					log.error( 'plugincollection-load: It was not possible to load the plugin.', { plugin: PluginConstructorOrName } );
```

Comment:
Do I see correctly that this error will be thrown if a plugin cannot be load because it cannot be found by its name? This should be a separate, more meaningful error.